### DISCUSSION TITLE
Titlebar customization - UWP

### DISCUSSION THREAD

**Discussion by hvansiaThor** [unoplatform/uno#14763] [Q&A]
Hello, I am trying to update the title of my application during runtime. 

I have tried couple of options:` Windows.UI.ViewManagement.ApplicationView.GetforCurrentView().Title `and `CoreApplication.GetCurrentView().TitleBar`. Both of them result in error. Uno documentation mentions about `Windows.UI.ViewManagement.ApplicationViewTitleBar` but there is no sample code showing/explaining how to use it. Can somebody point me to right direction/page which explains how title can be changed?

Thanks in advance!

**Comment by limeniye** [comment]
Hello, @hvansiaThor .
Are you sure that you are using the UWP project?
In theory, everything should work in UWP. Such problems can occur in WinUI.

In this case, as you can see the [ApplicationView is not supported](https://learn.microsoft.com/en-us/windows/apps/desktop/modernize/desktop-to-uwp-supported-api#apis-that-have-dependencies-on-uwp-only-ui-features) already.
The title bar [is not supported](https://github.com/unoplatform/uno/issues/14275) yet also.

As temporary, you can use the following way:
```C#
public sealed partial class MainPage : Page
{
    public MainPage()
    {
        this.InitializeComponent();
        Loaded += OnLoaded;
    }
    private void OnLoaded(object s, RoutedEventArgs e)
    {
#if WINDOWS
        var appWindowId = XamlRoot.ContentIslandEnvironment.AppWindowId;
        var titleBar = AppWindow.GetFromWindowId(appWindowId).TitleBar;
        titleBar.BackgroundColor = Colors.Transparent;
#endif
    }
}
```

For customization your TitleBar I suggest you to try the following sample: 
https://github.com/microsoft/WindowsAppSDK/discussions/3951#discussioncomment-7566571
